From 1eb3c109c3e891fabab5be0d5fa1014a99e6b4a6 Mon Sep 17 00:00:00 2001
From: Nate Drude <nate.d@variscite.com>
Date: Fri, 2 May 2025 11:38:24 -0500
Subject: [PATCH] imx93_var_som: Add mender configuration options

This patch enables the defconfig options expected by mender.
See: https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/recipes-bsp/u-boot/u-boot-mender.inc#L214-L268

Signed-off-by: Nate Drude <nate.d@variscite.com>
---
 configs/imx93_var_som_defconfig | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/configs/imx93_var_som_defconfig b/configs/imx93_var_som_defconfig
index bc75c68f980..053faa73693 100644
--- a/configs/imx93_var_som_defconfig
+++ b/configs/imx93_var_som_defconfig
@@ -7,7 +7,7 @@ CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
 CONFIG_NR_DRAM_BANKS=2
 CONFIG_SF_DEFAULT_SPEED=40000000
-CONFIG_ENV_SIZE=0x4000
+CONFIG_ENV_SIZE=0x20000
 CONFIG_ENV_OFFSET=0x700000
 CONFIG_IMX_CONFIG=""
 CONFIG_DM_GPIO=y
@@ -21,6 +21,7 @@ CONFIG_SPL_SERIAL=y
 CONFIG_SPL_DRIVERS_MISC=y
 CONFIG_SPL_STACK=0x20519dd0
 CONFIG_SPL=y
+CONFIG_ENV_OFFSET_REDUND=0x740000
 CONFIG_CMD_DEKBLOB=y
 CONFIG_SPL_IMX_ROMAPI_LOADADDR=0x88000000
 CONFIG_SYS_LOAD_ADDR=0x80400000
@@ -96,10 +97,9 @@ CONFIG_SPL_OF_CONTROL=y
 CONFIG_OF_LIST="imx93-var-som-symphony imx93-var-dart-dt8mcustomboard"
 CONFIG_MULTI_DTB_FIT=y
 CONFIG_ENV_OVERWRITE=y
-CONFIG_ENV_IS_NOWHERE=y
 CONFIG_ENV_IS_IN_MMC=y
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
-CONFIG_SYS_MMC_ENV_DEV=1
 CONFIG_USE_ETHPRIME=y
 CONFIG_ETHPRIME="eth0"
 CONFIG_NET_RANDOM_ETHADDR=y
@@ -107,6 +107,8 @@ CONFIG_SPL_DM=y
 CONFIG_SYSCON=y
 CONFIG_ADC=y
 CONFIG_ADC_IMX93=y
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
 CONFIG_SPL_CLK_IMX93=y
 CONFIG_CLK_IMX93=y
 CONFIG_CPU=y
-- 
2.49.0

